### DISCUSSION TITLE
[FR] Emoji support for markdown

### DISCUSSION THREAD

**Discussion by KatieMFritz** [wbrowar/craft-3-guide#62] [Ideas and Feature Requests]
Saving new Guides from the code editor errors out if you include any emojis. But emojis are very useful for documentation! 🎉

Error:
`General error: 1366 Incorrect string value: '\xF0\x9F\x98\x84\x0D\x0A...' for column 'content' at row 1`

**Comment by wbrowar** [comment]
Thanks for the suggestion, @KatieMFritz! I'm going to move this over to Discussions and share some good news there 😃

**Reply by wbrowar** [reply]
I want to test this out a little more, but I've hooked into the way Craft handles storing emoji and it works!

They use a library, called LitEmoji, to encode and decode emoji characters. In most cases it converts emoji so you’ll be able to view and edit them just fine, however, some of the newer emoji aren’t available just yet. In that case, you can still use the emoji, but they'll get converted to hex in the editor.

Editor:
<img width="885" alt="Screen Shot 2022-03-27 at 8 59 26 PM" src="https://user-images.githubusercontent.com/2158662/160310469-43767d0f-769c-470b-849c-ae8102454261.png">

Display:
<img width="980" alt="Screen Shot 2022-03-27 at 8 58 41 PM" src="https://user-images.githubusercontent.com/2158662/160310412-e9489e5d-1871-430a-b566-8b2e5800656d.png">
